google/cloud/spanner_dbapi/client_side_statement_executor.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 20203 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from google.cloud.spanner_dbapi.parsed_statement import (
+    ParsedStatement,
+    ClientSideStatementType,
+)
+
+
+class StatementExecutor(object):
+    def __init__(self, connection):
+        self.connection = connection
```

Comment:
Is `self.connection` intended to be a public property on this class?

I would naively assume that it'd be a private member.  In Python:

* Members that are intended to be private are, by convention, prefixed with an underscore (`self._connection`).  This leading underscore has no functional impact -- callers can still refer to the field with no problem, the interpreter doesn't treat it as special though some linters do -- but it's very obvious during a code review if someone is accessing a field with a leading underscore.
* Members that really need to be private are prefixed with two underscores (`self.__connection`).  In this case, Python will play namespace games to make it hard (though not impossible if someone tries hard enough) to access the field.  In practice, I can't recall having ever seen this used in production code; people just use the single underscore and accept that breaking the abstraction is an occasionally-necessary antipattern; but it's an option if you need it.

Reply:
Removed the class so not needed